src/users/users.service.ts

```diff
@@ -3,24 +3,138 @@ import { User } from '@prisma/client'
+  async findAll(
+    nameFilter?: string,
+    page?: number,
+    pageSize?: number,
+  ): Promise<UserPreview[]> {
+    const users = await this.prisma.user.findMany({
       where: {
         fullName: {
           contains: nameFilter ?? '',
           mode: 'insensitive',
         },
       },
+      include: {
+        _count: {
+          select: {
+            participations: true,
+          },
+        },
+        presentations: {
+          include: {
+            ratings: {
+              select: {
+                value: true,
+              },
+            },
+          },
+        },
+      },
+      take: pageSize || 20,
+      skip: (page || 0) * (pageSize || 20),
+    })
+
+    return users.map(({ presentations: p, _count: c, id, fullName }) => {
+      const ratings: number[] = p.reduce<number[]>(
+        (arr, pres) => [...arr, ...pres.ratings.map((r) => r.value)],
+        [],
+      )
+      const presentations = p.length
+      const attendances = c.participations
+      const avarageRating =
+        ratings.reduce((sum, rating) => sum + rating, 0) / ratings.length
+
+      return { presentations, attendances, avarageRating, id, fullName }
     })
   }
 
-  async findOne(id: number): Promise<User> {
-    return this.prisma.user.findUnique({ where: { id: id } })
+  async findOne(id: number): Promise<UserDetails> {
+    const user = await this.prisma.user.findUnique({
+      where: { id: id },
+      include: {
+        presentations: {
+          include: {
+            consultation: {
+              include: {
+                subject: true,
+              },
+            },
+            ratings: {
+              include: {
+                ratedBy: {
+                  include: {
+                    user: true,
+                  },
+                },
+              },
+            },
+          },
+        },
+        participations: {
+          include: {
+            consultation: {
+              include: {
+                subject: true,
+              },
+            },
+          },
+        },
+        requestedConsultations: {
+          include: {
+            subject: true,
+          },
+        },
+      },
+    })
+
+    const ratings: number[] = user.presentations.reduce<number[]>(
+      (arr, pres) => [...arr, ...pres.ratings.map((r) => r.value)],
+      [],
+    )
+    const avarageRating =
+      ratings.reduce((sum, rating) => sum + rating, 0) / ratings.length
+
+    const presentations = user.presentations.map(
+      ({ ratings, consultation }) => {
+        return {
+          ...consultation,
+          ratings: ratings
+            .filter((r) => !r.anonymous)
+            .map(({ ratedBy, ...rating }) => {
+              return {
+                ...rating,
+                rater: {
+                  id: ratedBy.userId,
+                  fullName: ratedBy.user.fullName,
+                },
+              }
+            }),
+        }
+      },
+    )
```

Comment:
I think we shouldn't filter out the anonymous ratings, we should just omit the user info. So the rater field should be optional in the DTO. Otherwise the most critical comments would be lost :D